src/typings/graphiql.d.ts

```diff
@@ -0,0 +1,21 @@
+declare module "graphiql" {
+  import { Component, ReactElement } from "react";
+  import { GraphQLSchema } from "graphql";
+  import { Observable } from "rxjs";
+
+  interface FetchConfig {
+    query: string;
+    variables?: object;
+  }
+
+  type Fetcher = (
+    config: FetchConfig
+  ) => Promise<unknown> | Observable<unknown> | unknown;
```

Comment:
is that something we chose or is that decided by `graphiQL`? 

if the latter, isn't this rather more like the following type? 

```typescript
type Fetcher<T> = (
  config: FetchConfig
) => Promise<T> | Observable<T>;
```

(i did not dig too much here, just want to make sure this has been a fully informed decision)

Reply:
You are right, I misread the docs